File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majority = 0;
        int count = 0;

        for(int num : nums) {
            if(count == 0) {
                majority = num;
                count = 1;
            } else {
                if(num == majority) {
                    count += 1;
                } else {
                    count -= 1;
                }
            }
        }
        return majority;
    }
};

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        new_d={}
        for i in nums:
            if i in new_d:
                new_d[i] += 1
            else:
                new_d[i] =1
        # a=new_d.values()
        a=max(new_d ,key=new_d.get)
        return a
